### ISSUE TITLE
Freature request: collapsible comments & adjustable post size

### ISSUE THREAD

**Issue by poVoq** [friendica/friendica#6690]
Would be nice if there was a per profile setting (like in Hubzilla) to have all comments hidden behind a comments button. This is especially nice for "news" channels or those re-posting an RSS feed etc.

It would be also nice if the size of the first post could be limited and you need to press "read more" or something like that to open up the full text. Ideally that should go down from a few lines of the text as preview all the way to just showing the title that can be clicked to open the message & comments.

I am guessing that this might also be good to have for the German "Leistungschutzrecht" and the new EU "link tax" for those wishing to repost RSS feeds or other sources?

Thanks for considering!

**Comment by MrPetovan** [comment]
There already a feature hiding all the top-level comments but the last three behind a button.

And there's an addon called "Show more" that collapse posts whose text is longer than a fixed amount of characters.

Do you need more specific features?

**Comment by poVoq** [comment]
Yeah, would be very good to be able to condense it further to zero comments.

Thanks for reminding me of the "show more" addon. It works, but the posts are still taking up a lot of space. Ideally I would like a compact view like Reddit for example.

**Comment by tobiasd** [comment]
I don'T think this should be a general option. For me this sounds like you want to write a new theme, special for your use-case ;-) It is not *that* complicated. And as far as I understand you, this should only need to touch the theming of the `network`, `display` and `profile` output.

@fabrixxm had once worked on a "Blog" theme with a close comment view for the profile page. Maybe you could use that theme as a starting point.

**Comment by poVoq** [comment]
Show comments or not seems rather a core feature and not something left to the theme. Otherwise I agree that of course :)

**Comment by MrPetovan** [comment]
A compact view like Reddit seems to be a rather different theme.

And currently we load all the comments even if we don't show all of them right away, so hiding the 3 left wouldn't improve performances.

I see two potential improvements axes here:
- Allow users to define the number of top-level comments then want not to be hidden on page load (it would be 0 in @poVoq 's case)
- Improve the `showmore` addon by using the post height as limit instead of an arbitrary number of characters, easily defeated by images for example.
